frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkTypeOptions.tsx

```diff
@@ -62,12 +62,13 @@ const NetworkTypeOptions = (props) => {
   const dynamicContent = _.map(params, (parameter, key) => {
     const validationMsg = _.get(typeParamsData, [key, 'validationMsg'], null);
     const elemType = _.get(parameter, 'type');
+    const value = _.get(typeParamsData, `${key}.value`, '');
```

Comment:
it looks like we convert `NetworkTypeParameter` to any and assuem it has a `value` field

if we do, we can simplify  `!_.isBoolean(value) ? parameter?.values?.checked : value` in L116 and just set value in the list and then use value as is in checkbox.

can you check if we actually have a hidden `value` filed in `NetworkTypeParameter`

Reply:
typeParamsData is clean in loading, it accumulates data only when the user starts to add it as typing, checking, etc. , so basically !_.isBoolean(value) ? parameter?.values?.checked : value , tells use the default value, once onChange run on first loading it will accumulate this value and then will use typeParamsData

Reply:
`parameter?.values` are the values for dropdown input options, they are not default values, they are a list of strings to populate the dropdown options.

Reply:
sorry in the list push its changed to !_.isBoolean(value) ? parameter?.initValue : value maybe now its more clear